modules/nextflow/src/main/groovy/nextflow/splitter/JsonSplitter.groovy

```diff
@@ -0,0 +1,287 @@
+/*
+ * Copyright 2013-2023, Seqera Labs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package nextflow.splitter
+
+import com.google.gson.stream.JsonReader
+import com.google.gson.stream.JsonToken
+import groovy.transform.CompileStatic
+import groovy.transform.InheritConstructors
+import groovy.util.logging.Slf4j
+
+/**
+ * Split a JSON document into records
+ *
+ * @author Pierre Lindenbaum PhD Institut-du-Thorax Nantes France.
+ */
+@Slf4j
+@CompileStatic
+@InheritConstructors
+class JsonSplitter extends AbstractTextSplitter {
+
+    public static final String OBJECT_KEY = 'key'
+    public static final String OBJECT_VALUE = 'value'
+
+    /**
+     * json path to find a specific region of the json
+     */
+    private String jsonPath
+
+    /**
+     * number of items processed so far
+     */
+    private long itemsCount = 0
+
+    /**
+     * Set the splitter options by specifying a map of named parameters.
+     * Valid parameters are:
+     * <li>{@code jsonPath}
+     *
+     * @param options
+     * @return The splitter instance itself
+     */
+    @Override
+    JsonSplitter options(Map opts) {
+        super.options(opts)
+        if (opts.jsonPath) {
+            this.jsonPath = (opts.jsonPath as String)
+        }
+        return this
+    }
+
+    /**
+     * @return A map representing the valid options for the splitter. The map keys define the
+     * accepted parameter names, the values the valid values for each of them.
+     */
+    @Override
+    protected Map<String,?> validOptions() {
+        def baseOptions = super.validOptions()
+        return [
+            jsonPath: String,
```

Comment:
```suggestion
            path: String,
```

Using `path` is a bit more concise and readable e.g. `something.splitJson(path:'x.y.z')`